Rubberduck.Parsing/ComReflection/ComVariant.cs

```diff
@@ -59,12 +88,79 @@ private struct Variant
 
         public ComVariant(IntPtr variant)
         {
-            Value = Marshal.GetObjectForNativeVariant(variant);
             var members = Marshal.PtrToStructure<Variant>(variant);
             VariantType = (VarEnum)members.vt;
-            if (Value == null && VariantType == VarEnum.VT_BSTR)
+
+            // Note that these are technically flags, but it they are VT_BYREF or VT_ARRAY, the data area will always be a pointer.
+            // ReSharper disable once SwitchStatementMissingSomeCases
+            switch ((VarEnum)members.vt)
             {
-                Value = string.Empty;
+                case VarEnum.VT_VOID:
+                case VarEnum.VT_EMPTY:
+                case VarEnum.VT_NULL:
+                case VarEnum.VT_ERROR:
```

Comment:
Just checking - are we losing out any interesting information for the COM collector (e.g. default value for optional parameters?) here?

Reply:
I'm not exactly sure what you would pass in VBA as an equivalent if you were explicitly passing in the default other than `Null`.